### PR TITLE
sync script - never ask gclient to sync brave-core

### PR DESCRIPTION
**Pull request by petemill**
Resolves https://github.com/brave/brave-browser/issues/20985

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: build/commands/lib/util.js [hunk 1]
```diff
@@ -704,8 +673,7 @@ const util = {
     restoreBraveCoreGitUrlIfGitCacheEnabled()
```
Comment (goodov) [inline build/commands/lib/util.js:673]: we can drop this as well with this PR.

File: build/commands/scripts/sync.js [hunk 2]
```diff
@@ -12,7 +12,6 @@ const os = require('os')
 
 program
   .version(process.env.npm_package_version)
-  .arguments('[ref]')
```
Comment (atuchin-m) [inline build/commands/scripts/sync.js:15]: We need to also update the docs:
https://github.com/brave/brave-browser#update-brave

**Review by atuchin-m — APPROVED** [review]

**Comment by mihaiplesa** [comment]
Would this prevent the below?

```
09:46:38  /home/ubuntu/workspace/test-brave-browser-build-linux-x64-b-c-nightly/src/brave
09:46:38  > git checkout c6c97a29225b188ef87b815393d6945e748fcaef
09:46:38  ...brave core is now at commit ID c6c97a29225b188ef87b815393d6945e748fcaef
09:46:38  using git cache path /home/ubuntu/cache
09:46:38  -------------------------------
09:46:38  /home/ubuntu/workspace/test-brave-browser-build-linux-x64-b-c-nightly
09:46:38  > gclient sync --reset --nohooks --revision src/brave@c6c97a29225b188ef87b815393d6945e748fcaef --revision src@refs/tags/97.0.4692.36 --force --with_tags --with_branch_heads --upstream
09:46:56  WARNING: Your metrics.cfg file was invalid or nonexistent. A new one will be created.
09:46:56  Error: Invalid .gclient file. Solution is incomplete: {'managed': False, 'name': 'src/brave'}
```

**Comment by atuchin-m** [comment]
One extra thing : `url: config.braveCoreRepo` should be removed from `.gclient` shouldn't it?

**Review by bridiver — APPROVED** [review]

File: build/commands/lib/util.js (hunk 1 above)
Comment (goodov) [reply build/commands/lib/util.js:688]: @petemill please remove this before merging. I've added this workaround for the exact case we're removing here - when brave-core was synced via `gclient`.

Comment (mihaiplesa) [reply build/commands/lib/util.js:688]: @goodov got a link to the workaround?

File: build/commands/lib/util.js [hunk 3]
```diff
@@ -287,20 +289,33 @@ const util = {
...
-            Math.floor(new Date(fs.statSync(sourceFile).mtimeMs).getTime() / 1000) !=
-            Math.floor(new Date(fs.statSync(destinationFile).mtimeMs).getTime() / 1000)) {
-          fs.copySync(sourceFile, destinationFile)
-          // can't set the date in the past so update the source file
-          // to match the newly copied destionation file
-          const date = fs.statSync(destinationFile).mtime
-          fs.utimesSync(sourceFile, date, date)
-          console.log(sourceFile + ' copied to ' + destinationFile)
+        let sourceDate, destinationDate
+        if (fs.existsSync(destinationFile)) {
```
Comment (goodov) [inline build/commands/lib/util.js:293]: why don't we use `isOverrideNewer` here?

File: build/commands/lib/util.js [hunk 4]
```diff
@@ -287,20 +289,33 @@ const util = {
...
-        if (!fs.existsSync(destinationFile) ||
-            Math.floor(new Date(fs.statSync(sourceFile).mtimeMs).getTime() / 1000) !=
-            Math.floor(new Date(fs.statSync(destinationFile).mtimeMs).getTime() / 1000)) {
-          fs.copySync(sourceFile, destinationFile)
-          // can't set the date in the past so update the source file
-          // to match the newly copied destionation file
-          const date = fs.statSync(destinationFile).mtime
-          fs.utimesSync(sourceFile, date, date)
-          console.log(sourceFile + ' copied to ' + destinationFile)
+        let sourceDate, destinationDate
```
Comment (bsclifton) [inline build/commands/lib/util.js:292]: these are defined just below; I think we can delete them

File: build/commands/lib/util.js (hunk 3 above)
Comment (bsclifton) [reply build/commands/lib/util.js:293]: maybe even just move `isOverrideNewer` here into utils 👍 (and also use it here)

**Review by bsclifton — COMMENTED** [review]
Left comments; besides those, LGTM 😄👍

**Comment by petemill** [comment]
Just to be clear, a lot of the code currently in this PR is throwaway code for explicitly logging issues we're having in CI during the validation of the main code. When we have sync working correctly on all platforms (seems like Linux is the only issue we have right now) then I'll remove the extra logging and re-tag for review.

File: build/commands/lib/util.js (hunk 3 above)
Comment (petemill) [reply build/commands/lib/util.js:293]: This is temporary code copy pasted for extra logging whilst we're having issues in CI with this approach. This Pr is currently in exploitative state (wild guessing using code and extra logging)

**Comment by mihaiplesa** [comment]
There are intermittent issues like the below.

```
powershell.exe : From https://github.com/brave/brave-core[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-2)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-3)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-4)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-5)

[2022-02-11T21:46:43.624Z] At C:\ws@tmp\durable-29f250c5\powershellWrapper.ps1:3 char:1

[2022-02-11T21:46:43.624Z] + & powershell -NoProfile -NonInteractive -ExecutionPolicy Bypass -Comm ...

[2022-02-11T21:46:43.624Z] + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

[2022-02-11T21:46:43.624Z]     + CategoryInfo          : NotSpecified: (From https://github.com/brave/brave-core:String[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-6)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-7)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-8)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-9)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-10)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-11)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-12)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-13)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-14)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-15)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-16)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-17)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-18)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-19)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-20)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-21)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-22)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-23)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-24)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-25)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-26)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-27)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-28)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-29)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-30)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-31)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-32)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-33)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-34)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-35)) [], RemoteException

[2022-02-11T21:46:43.624Z]     + FullyQualifiedErrorId : NativeCommandError

[2022-02-11T21:46:43.624Z]  

[2022-02-11T21:46:43.624Z]    7a5fa5ab51..fe042358ac  bls-integration        -> origin/bls-integration

[2022-02-11T21:46:43.624Z]  + 08e3f12836...87775239f3 bsc-vpn-remaining-work -> origin/bsc-vpn-remaining-work  (forced update)

[2022-02-11T21:46:43.624Z]  + 6dbbac170e...a6edeb6c8b cr99                   -> origin/cr99  (forced update)

[2022-02-11T21:46:43.624Z]    3986a06a40..1acbad17d3  emerick-cr100          -> origin/emerick-cr100

[2022-02-11T21:46:43.624Z]  * [new branch]            wallet-usage-fix2      -> origin/wallet-usage-fix2

[2022-02-11T21:46:47.803Z] HEAD is now at 4e0dba65f1 Update Jenkinsfile

[2022-02-11T21:46:48.059Z] 

[2022-02-11T21:46:48.059Z] > brave-core@1.37.43 sync

[2022-02-11T21:46:48.059Z] > node ./build/commands/scripts/sync.js

[2022-02-11T21:46:48.059Z] 

[2022-02-11T21:46:51.333Z] Brave Browser Sync starting

[2022-02-11T21:46:51.333Z] Running gclient sync...

[2022-02-11T21:46:51.333Z] -------------------------------

[2022-02-11T21:46:51.333Z] C:\ws\src

[2022-02-11T21:46:51.333Z] > git rev-parse HEAD

[2022-02-11T21:46:51.333Z] -------------------------------

[2022-02-11T21:46:51.333Z] C:\ws\src

[2022-02-11T21:46:51.333Z] > git rev-parse refs/tags/98.0.4758.87

[2022-02-11T21:46:51.333Z] Chromium repo does not need update as it is already refs/tags/98.0.4758.87 at commit e5d25fcbcb550c86e6390774452f51cf75261624.

[2022-02-11T21:46:51.333Z] using git cache path C:/Users/Administrator/cache

[2022-02-11T21:46:51.333Z] -------------------------------

[2022-02-11T21:46:51.333Z] C:\ws

[2022-02-11T21:46:51.333Z] > gclient sync --nohooks

[2022-02-11T21:47:17.854Z] 

[2022-02-11T21:47:17.854Z] src\brave\vendor\omaha (ERROR)

[2022-02-11T21:47:17.854Z] ----------------------------------------

[2022-02-11T21:47:17.854Z] [0:00:11] Started.

[2022-02-11T21:47:17.854Z] [0:00:11] Finished running: git remote set-url --push origin https://github.com/brave/omaha.git[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-36)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-37)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-38)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-39)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-40)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-41)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-42)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-43)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-no-gclient-brave-core-windows/detail/pr-brave-browser-no-gclient-brave-core-windows/34/pipeline#step-153-log-44)

[2022-02-11T21:47:17.854Z] running "git cat-file -e a199d4a386798c4ecc1a29f1a80f58a1da4755ff^^^^{commit}" in "C:\Users\Administrator\cache\github.com-brave-omaha"

[2022-02-11T21:47:17.854Z] [0:00:12] Finished running: git config remote.origin.url

[2022-02-11T21:47:17.854Z] [0:00:12] Finished running: git rev-list -n 1 HEAD

[2022-02-11T21:47:17.854Z] [0:00:12] Finished running: git rev-parse --abbrev-ref=strict HEAD

[2022-02-11T21:47:17.854Z] ----------------------------------------

[2022-02-11T21:47:17.854Z] Error: 119> 

[2022-02-11T21:47:17.854Z] 119> ____ src\brave\vendor\omaha at a199d4a386798c4ecc1a29f1a80f58a1da4755ff

[2022-02-11T21:47:17.854Z] 119> 	You have unstaged changes.

[2022-02-11T21:47:17.854Z] 119> 	Please commit, stash, or reset.
```

**Comment by mihaiplesa** [comment]
@mherrmann @simonhong any idea while trying to reuse the workspace of Windows would produce the above error?

**Comment by mherrmann** [comment]
This likely happens since #11970 was merged. There, we removed the `--reset` flag in the call to `gclient sync --nohooks`. A very quick fix / workaround would be to temporarily re-introduce the `--reset` flag.

In this particular case, my suspicion is that the "unstaged changes" in `.../omaha/` are modifications made to `src/brave/vendor/omaha/omaha/standalone/standalone_installers.txt` by [`build_omaha.py:100`](https://github.com/brave/omaha/blob/3047695e134fb4bd38d1f489721404b97cb09067/build_omaha.py#L100).

In general, it's possible we will also run into other, similar issues with other dependencies.

**Comment by petemill** [comment]
The removal of reset was aimed to help sync have less to re-do. Is there a fix we can do before this PR is merged that isn't putting reset back in? Is the Omaha issue a brave or chromium issue?

**Comment by mherrmann** [comment]
@petemill the fix is to modify github.com/brave/omaha to not have unstaged changes after a normal build. I will look into it from tomorrow, now that @mihaiplesa has agreed.

**Comment by mherrmann** [comment]
I created the [issue](https://github.com/brave/brave-browser/issues/21103) as discussed and am beginning work on a fix.

**Comment by mherrmann** [comment]
I have a fix attempt in https://github.com/brave/brave-core/pull/12270 and am looking for reviewers to approve it.

**Comment by mherrmann** [comment]
My PR was merged. @petemill you could try to rebase this PR and see if you still get errors from unstaged changes in `src/brave/vendor/omaha`.

**Comment by petemill** [comment]
> My PR was merged. @petemill you could try to rebase this PR and see if you still get errors from unstaged changes in `src/brave/vendor/omaha`.

Just noticed this - that's great. Let's see what happens from the current build

**Review by mihaiplesa — APPROVED** [review]

**Comment by mihaiplesa** [comment]
This is ready to go, please re-review.

**Comment by atuchin-m** [comment]
Look good to me.

File: build/commands/lib/util.js (hunk 1 above)
Comment (mihaiplesa) [reply build/commands/lib/util.js]: Removed in https://github.com/brave/brave-core/pull/11543/commits/1aef34849c0e6fa8df47f013d1081a5b6a38a15f

**Comment by mihaiplesa** [comment]
```
13:57:36  /home/ubuntu/workspace/pr-brave-browser-no-gclient-brave-core-android
13:57:36  > gclient sync --nohooks
13:59:57  
13:59:57  src/third_party/icu (ERROR)
13:59:57  ----------------------------------------
13:59:57  [0:00:06] Started.
13:59:57  [0:00:06] Finished running: git remote set-url --push origin https://chromium.googlesource.com/chromium/deps/icu.git
13:59:57  running "git cat-file -e 609143a639dcc0884a87c899614895ce129afdb8^{commit}" in "/home/ubuntu/cache/chromium.googlesource.com-chromium-deps-icu"
13:59:57  [0:00:07] Finished running: git config remote.origin.url
13:59:57  [0:00:08] Finished running: git rev-list -n 1 HEAD
13:59:57  [0:00:09] Finished running: git rev-parse --abbrev-ref=strict HEAD
13:59:57  ----------------------------------------
13:59:57 
 Error: 252> 
13:59:57  252> ____ src/third_party/icu at 609143a639dcc0884a87c899614895ce129afdb8
13:59:57  252> 	You have unstaged changes.
13:59:57  252> 	Please commit, stash, or reset.
```

**Comment by mihaiplesa** [comment]
@brave/android any idea why this happen on sync for Android?

```
13:57:36  /home/ubuntu/workspace/pr-brave-browser-no-gclient-brave-core-android
13:57:36  > gclient sync --nohooks
13:59:57  
13:59:57  src/third_party/icu (ERROR)
13:59:57  ----------------------------------------
13:59:57  [0:00:06] Started.
13:59:57  [0:00:06] Finished running: git remote set-url --push origin https://chromium.googlesource.com/chromium/deps/icu.git
13:59:57  running "git cat-file -e 609143a639dcc0884a87c899614895ce129afdb8^{commit}" in "/home/ubuntu/cache/chromium.googlesource.com-chromium-deps-icu"
13:59:57  [0:00:07] Finished running: git config remote.origin.url
13:59:57  [0:00:08] Finished running: git rev-list -n 1 HEAD
13:59:57  [0:00:09] Finished running: git rev-parse --abbrev-ref=strict HEAD
13:59:57  ----------------------------------------
13:59:57 
 Error: 252> 
13:59:57  252> ____ src/third_party/icu at 609143a639dcc0884a87c899614895ce129afdb8
13:59:57  252> 	You have unstaged changes.
13:59:57  252> 	Please commit, stash, or reset.
```

**Comment by SergeyZhukovsky** [comment]
> @brave/android any idea why this happen on sync for Android?
> 
> ```
> 13:57:36  /home/ubuntu/workspace/pr-brave-browser-no-gclient-brave-core-android
> 13:57:36  > gclient sync --nohooks
> 13:59:57  
> 13:59:57  src/third_party/icu (ERROR)
> 13:59:57  ----------------------------------------
> 13:59:57  [0:00:06] Started.
> 13:59:57  [0:00:06] Finished running: git remote set-url --push origin https://chromium.googlesource.com/chromium/deps/icu.git
> 13:59:57  running "git cat-file -e 609143a639dcc0884a87c899614895ce129afdb8^{commit}" in "/home/ubuntu/cache/chromium.googlesource.com-chromium-deps-icu"
> 13:59:57  [0:00:07] Finished running: git config remote.origin.url
> 13:59:57  [0:00:08] Finished running: git rev-list -n 1 HEAD
> 13:59:57  [0:00:09] Finished running: git rev-parse --abbrev-ref=strict HEAD
> 13:59:57  ----------------------------------------
> 13:59:57 
>  Error: 252> 
> 13:59:57  252> ____ src/third_party/icu at 609143a639dcc0884a87c899614895ce129afdb8
> 13:59:57  252> 	You have unstaged changes.
> 13:59:57  252> 	Please commit, stash, or reset.
> ```

@mihaiplesa does it happen on `npm run sync`? I don't remember I ever seen that on my local builds. cc @samartnik as well.

**Comment by mihaiplesa** [comment]
@SergeyZhukovsky we're trying to switch away from init to sync and the above appears on succesive sync. I don't remember seeing it before when testing.

**Comment by wchen342** [comment]
It has something to do with `custom_deps` I believe. This happens when there is a modified `.gclient` and when `fetch` and `sync` and done separately. Maybe a `git diff` inside `src/third_party/icu` will give more insights?

**Comment by mihaiplesa** [comment]
@wchen342 

```
09:24:17  + git -C src/third_party/icu diff
09:24:17  diff --git a/android_small/icudtl.dat b/android_small/icudtl.dat
09:24:17  old mode 100644
09:24:17  new mode 100755
09:24:17  diff --git a/android_small/icudtl_extra.dat b/android_small/icudtl_extra.dat
09:24:17  old mode 100644
09:24:17  new mode 100755
09:24:17  diff --git a/common/icudtl.dat b/common/icudtl.dat
09:24:17  old mode 100644
09:24:17  new mode 100755
```

**Review by goodov — APPROVED** [review]

**Comment by mihaiplesa** [comment]
```
17:05:37  228> ____ src/third_party/devtools-frontend/src at 1655fc6e5b1b7db015e2b0fc9dfe75f6b0034388
17:05:37  228> 	You have unstaged changes.
17:05:37  228> 	Please commit, stash, or reset.
```

**Comment by mihaiplesa** [comment]
We are good to merge now.

**Review by atuchin-m — APPROVED** [review]

**Review by goodov — APPROVED** [review]

**Comment by petemill** [comment]
I will squash and clean before merge

**Review by bridiver — APPROVED** [review]

File: build/commands/scripts/sync.js (hunk 2 above)
Comment (mihaiplesa) [reply build/commands/scripts/sync.js]: This is still outstanding @petemill.

**Pull request merged**